src/nnue/nnue_feature_transformer.h

```diff
@@ -247,14 +250,22 @@ class FeatureTransformer {
         return !stream.fail();
     }
 
+    // Cast a pointer to a 2 dimensional array of width D
+    template<int D, typename T>
+    static constexpr T (*cast_2D(T* pt))[D] {
+        return (T(*)[D]) pt;
+    }
+
     // Convert input features
     std::int32_t transform(const Position& pos, OutputType* output, int bucket) const {
         update_accumulator<WHITE>(pos);
         update_accumulator<BLACK>(pos);
 
-        const Color perspectives[2]  = {pos.side_to_move(), ~pos.side_to_move()};
-        const auto& accumulation     = pos.state()->accumulator.accumulation;
-        const auto& psqtAccumulation = pos.state()->accumulator.psqtAccumulation;
+        const Color perspectives[2] = {pos.side_to_move(), ~pos.side_to_move()};
+        const auto& accumulation =
+          cast_2D<TransformedFeatureDimensions>(pos.state()->template accumulation<Small>());
```

Comment:
perhaps this should be resolved by a member pointer to the accumulator passed as a template parameter to this class instead? This class really shouldn't know about all the different networks available.

Reply:
I'm not sure I completely follow or know how to do this.

Reply:
https://godbolt.org/z/sWExj7bd1

Reply:
This really reduced the number of changes in nnue_feature_transformer.h   Thank you for the example code!